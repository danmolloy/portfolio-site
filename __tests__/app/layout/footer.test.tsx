import Footer from "@/app/layout/footer";
import { act, fireEvent, render, screen } from "@testing-library/react";



describe("<Footer />", () => {
  beforeEach(() => {
    render(<Footer />)
  })
  it("footer is in the document", () => {
    const footer = screen.getByTestId("footer")
    expect(footer).toBeInTheDocument()
  })
  it("back to top btn is in the document and calls scrollToTop on click", () => {
    const backToTopBtn = screen.getByText("Back to top")
    expect(backToTopBtn).toBeInTheDocument()
    expect(backToTopBtn).toHaveAttribute("href", "/")
    
  })
  it("design and building credit is in the document", () => {
    const buildCredit = screen.getByText("Designed and built by Daniel Molloy")
    expect(buildCredit).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const footer = screen.getByTestId("footer")
    expect(footer).toMatchSnapshot()
  })
})