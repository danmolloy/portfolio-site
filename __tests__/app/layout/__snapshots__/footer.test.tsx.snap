// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<div
  class="w-full h-20 dark:bg-gray-900 flex sm:flex-row flex-col sm:justify-between p-2 items-center border-t dark:border-t-zinc-600"
  data-testid="footer"
>
  <a
    class=" hover:underline text-sm font-light hover:cursor-pointer active:text-zinc-400 m-1 "
    href="/"
  >
    Back to top
  </a>
  <p
    class="text-sm  text-zinc-500 dark:text-zinc-400"
  >
    Designed and built by Daniel Molloy
  </p>
</div>
`;
